account_asset_management/models/account_asset.py

```diff
@@ -471,6 +471,13 @@ def group_lines(x, y):
 
         line_obj = self.env['account.asset.line']
         digits = self.env['decimal.precision'].precision_get('Account')
+        if 'fiscalyear_lock_date' in self.env.context:
```

Comment:
Why do you need to check this in the context? Go directly to the company data

Reply:
It's used only for tests (`test_00_fiscalyear_lock_date_month` + `test_00_fiscalyear_lock_date_year`).

Reply:
We shouldn't introduce such mechanism, but use regular one (field company) for that.